pkg/assets/types.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2022 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package assets
+
+// Asset types used throughout the system.
+const (
+	BootableDisk       = "bootable_disk"
+	NonBootableDisk    = "non_bootable_disk"
+	KernelObject       = "kernel_object"
+	VmLinux            = "vmlinux"
```

Comment:
Do we really need to make these types as generic as possible? `vmlinux` anyway looks like a totally different file type from files that play the same role for other OSes, so from the user perspective it'd be more clear if they were also designated differently in an email. I.e. `vmlinux file: %URL%` looks better than `kernel debug object: %URL%`.

Reply:
> Do we really need to make these types as generic as possible?

It depends on what are the reasons to make it non-generic. If there are no major reasons to make code non-generic, it's always better to have it generic from multiple perspectives.

Mapping them to different user-readable strings for different OSes does not look like a sufficient reason to make code non-generic on the protocol level.

We already have a notion of "kernel object" and can be different files for different OSes:
https://github.com/google/syzkaller/blob/1434eec0b84075b7246560cfa89f20cdb3d8077f/sys/targets/targets.go#L75-L76

These are not really totally different file types, these are all ELF files. And we have generic code that runs nm/objdump on them. If we save raw coverage, I assume we may want to have the same common handling for the kernel object files stored in the asset storage.

You already have the function that maps these types to user-readable strings, it would be reasonable to make that function to accept OS/target and produce different user-readable strings if that's what we want. Btw, maybe it could use the sys/targets.KernelObject names for this since these should be familiar to kernel developers.

Reply:
Thanks for the clarification, fixed.